### PR TITLE
refactor(eval_hub): derive MLflow InsecureSkipVerify from TLSConfig

### PR DESCRIPTION
**Pull request by julpayne**
## What and why

- Remove use of `InsecureSkipVerify` in the mlflow client - tests can provide their own (insecure) TLS config if really needed.
- Sidecar will never support `InsecureSkipVerify` with the mlflow client.
- Hardens sidecar request routing to strip query parameters and fragments when url.Parse rejects control characters in the URI.
- Fixes a failing test as well as a randomly failing (ociclient) test.

Assisted-by: Cursor

## Type

- [ ] feat
- [x] fix
- [ ] docs
- [x] refactor / chore
- [x] test / ci

## Testing

- [x] Tests added or updated
- [x] Tested manually

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved MLflow TLS configuration handling and certificate verification behavior.
  - Made sidecar request routing reliable when URLs contain malformed query or fragment characters.
  - Preserved MLflow connection settings when generating sidecar configurations.

- **Testing**
  - Expanded coverage for TLS behavior, sidecar configuration, routing edge cases, and HTTP client authentication.

- **Chores**
  - Updated the MLflow test environment default Python version to 3.12.

### CODE REVIEW

**Comment by julpayne** [comment]
@coderabbitai /review

**Comment by julpayne** [comment]
@coderabbitai /review

File: internal/eval_runtime_sidecar/config/sidecar_runtime_config.go [hunk 1]
```diff
@@ -45,12 +45,11 @@ func LoadSidecarRuntimeConfig(sidecarJSONPath, version, build, buildDate string)
...
-			HTTPTimeout:        sc.MLFlow.HTTPTimeout,
-			CACertPath:         sc.MLFlow.CACertPath,
-			InsecureSkipVerify: sc.MLFlow.InsecureSkipVerify,
-			TokenPath:          sc.MLFlow.TokenPath,
-			Workspace:          sc.MLFlow.Workspace,
+			TrackingURI: strings.TrimSpace(sc.MLFlow.TrackingURI),
+			HTTPTimeout: sc.MLFlow.HTTPTimeout,
+			CACertPath:  sc.MLFlow.CACertPath,
+			TokenPath:   sc.MLFlow.TokenPath,
+			Workspace:   sc.MLFlow.Workspace,
```
[the remark this answers is not in the document]
Comment (julpayne) [reply internal/eval_runtime_sidecar/config/sidecar_runtime_config.go]: Fixed

File: tests/mlflow/Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # Local Python environment (uv). Isolated from the repo-root .venv used for FVT.
 VENV_DIR ?= .venv
-PYTHON ?= 3.14
+PYTHON ?= 3.12
```
[the remark this answers is not in the document]
Comment (julpayne) [reply tests/mlflow/Makefile]: Fixed

**Comment by julpayne** [comment]
Re-doing this PR.
